rocketpy/Components.py

```diff
@@ -0,0 +1,125 @@
+__author__ = "Mateus Stano Junqueira"
+__copyright__ = "Copyright 20XX, RocketPy Team"
+__license__ = "MIT"
+
+from collections import namedtuple
+
+
+class Components:
+    """A class to hold components of the Rocket class. A component is
+    an object that added to the rocket given a position relative to the
+    rocket's coordinate system origin. Components can be added to the
+    rocket using the 'add' methods. This class is currently used only to
+    hold aerodynamic surfaces.
+
+    Attributes
+    ----------
+    _components: list
+        A list of named tuples containing the component and its position.
+    component_tuple: namedtuple
+        A named tuple containing the component and its position.
+    """
+
+    def __init__(self):
+        """Initialize an empty components list instance."""
+        self.component_tuple = namedtuple("component_tuple", "component position")
+        self._components = []
+
+    def __repr__(self):
+        """Return a string representation of the Components instance."""
+        return repr(self._components)
+
+    def __len__(self):
+        """Return the number of components in the collection."""
+        return len(self._components)
+
+    def __getitem__(self, index):
+        """Return the number of components in the collection."""
+        return self._components[index]
+
+    def __iter__(self):
+        """Return an iterator over the components."""
+        return iter(self._components)
+
+    def add(self, component, position):
+        """Add a component to the collection.
+
+        Parameters
+        ----------
+            component: AeroSurface, Motor
```

Comment:
You defined in the class' docstring that "this is currently used only for aerodynamic surfaces".
1 - Well, should we modify the class docstring or the add method?
2 - What other objects could possible be components in the future?
    - Rail buttons for sure
    - Maybe parachutes should be too?
    - Any other?

Reply:
Btw these lines are out of identation. This will break the docs if we merge like this.

Reply:
> You defined in the class' docstring that "this is currently used only for aerodynamic surfaces".
> 1 - Well, should we modify the class docstring or the add method?
> 2 - What other objects could possible be components in the future?
> - Rail buttons for sure
> - Maybe parachutes should be too?
> - Any other?

It could some day be used for other stuff. I guess I could change the documentation to be `any` instead of specifying a type. Because it really could be anything

Reply:
Keep in mind that every time we change the documentation, we are actually creating a breaking change,
So good idea to be flexible by now ;)

Reply:
I'm re-opening this conversation as I'm not sure if we are allowed to simply put "any" as the argument type requirement.

@MateusStano have you seen this approach in any other place?

I understand and agree that we should be flexible, but this seems new and tricky to me, I'm not sure if everybody would understand it.

Reply:
Yes I have seen this in pydantic and actually good catch because the `a` should be capitalized (e.g. `Any`)

Reply:
Example from https://docs.python.org/3/library/typing.html

![image](https://github.com/RocketPy-Team/RocketPy/assets/69485049/086857e5-88ce-4f7f-bb88-59cf6b5c5ca1)

Reply:
Done in https://github.com/RocketPy-Team/RocketPy/pull/355/commits/cec534a57655859b4c7aa47e372a2f051bd4312b